kernel/src/table_configuration.rs

```diff
@@ -126,10 +126,24 @@ impl TableConfiguration {
     }
     /// Returns `true` if the kernel supports writing to this table. This checks that the
     /// protocol's writer features are all supported.
-    #[allow(unused)]
     #[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
-    pub(crate) fn is_write_supported(&self) -> bool {
-        self.protocol.ensure_write_supported().is_ok()
+    pub(crate) fn ensure_write_supported(&self) -> DeltaResult<()> {
+        self.protocol.ensure_write_supported()?;
+
+        // for now we don't support invariants so although we allow writer version 2 and the
```

Comment:
```suggestion
        // for now we don't allow invariants so although we support writer version 2 and the
```
("support" has a very specific meaning in the Delta spec, and we _do_ technically support the table feature)